### PR TITLE
Issue #12. Delete items from my shopping list

### LINKED ISSUE #12: 12. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future
## Summary
Users might make a mistake when entering an item, or may decide they won’t be buying a certain item again in the future. Allowing them to delete list items will help them keep their lists tidy.
 
## Acceptance criteria
- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Notes:
* Activating the “Delete” button should show a confirmation dialog. We want to give users a chance to confirm “destructive” actions.
* It might be tempting to create your own modal dialog component to handle confirmation, but modals are fraught with accessibility concerns, so consider using a simple [JavaScript confirm dialog](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm).

### PR DESCRIPTION
**Pull request by chungleee**
## Description

This code allows user to click on a item to delete from database. It will trigger a confirmation dialog window and act accordingly.

## Related Issue

closes #12 

## Acceptance Criteria

- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Type of Changes

- [x] Enhancement

### Before
Screenshot 2024-03-11 at 12 27 43 PM

### After
Screenshot 2024-03-11 at 3 29 09 PM

Screenshot 2024-03-11 at 3 29 14 PM

## Testing Steps / QA Criteria

1. `git pull origin gl-lc-delete-item`
2. `git checkout gl-lc-delete-item`
3. Select a list
4. Go to `List` page and press delete on an item

### CODE REVIEW

**Review by MiliMade — APPROVED** [review]
Everything works for me! 

PeachDancingGIF

**Review by amandaguan-ag — APPROVED** [review]
The delete feature working well~

File: src/api/firebase.js [hunk 1]
```diff
@@ -224,7 +225,10 @@ export async function updateItem(listPath, itemId, dateLastPurchased) {
 	});
 }
 
-export async function deleteItem() {
+export async function deleteItem(listPath, itemId) {
+	const listCollectionRef = doc(db, `${listPath}/items`, itemId);
+
+	await deleteDoc(listCollectionRef);
 	/**
```
Comment (djtaylor8) [inline src/api/firebase.js]: Looks like we can remove this comment

File: src/components/ListItem.jsx [hunk 2]
```diff
@@ -16,6 +16,18 @@ export function ListItem({ name, listPath, itemId, dateLastPurchased }) {
 		}
 	};
 
+	const handleClick = async () => {
+		if (window.confirm('Do you really want to delete this item?')) {
+			try {
+				await deleteItem(listPath, itemId);
+			} catch (error) {
```
Comment (djtaylor8) [inline src/components/ListItem.jsx:24]: Should we display an error to the user here?

**Review by djtaylor8 — APPROVED** [review]
Simple and effective. Great work @g-lee2 and @chungleee!

**Review by RachaelSkye — APPROVED** [review]
Looks good! :shipit:

**Pull request merged**